docs/spec_draft.md

```diff
@@ -1350,6 +1352,189 @@ specification. Numeric precision is implementation-defined.
+
+### Outputs
+
+| Name     | Type                                              |
+|----------|---------------------------------------------------|
+| `results` | variadic number of tensors of any supported types|
+
+### Constraints
+
+
+  * (C1) For any k $\in$ [0, N), rank(`inputs`[k]) = size(`update_window_dims`) +
+         size(`inserted_window_dims`).
+
+  * (C2) N $=$ size(`inputs`) == size(`updates`) and N $\ge$ 1.
+
+  * (C3) All `inputs` have the same shape.
+
+  * (C4) All `updates` have the same shape.
+
+  * (C5) $0 \le$ `index_vector_dim` $\le$ rank(`scatter_indices`) Or
+        `index_vector_dim` $=$ rank(`scatter_indices`), in which case assume a
+         trailing $1$ on the shape of `scatter_indices`.
+
+  * (C6) dim(`scatter_indices`, `index_vector_dim`) $=$
+         size(`scatter_dims_to_operand_dims`)
+
+  * (C7) All dimensions in `scatter_dims_to_operand_dims` unique.
+
+  * (C8) For all i $\in$ [0, size(`scatter_dims_to_operand_dims`)), $0 \le$
+        `scatter_dims_to_operand_dims`[i] $\lt$ rank(`operand`[k]), for any k
+        $\in$ [0, N).
+
+  * (C9) All dimensions in `update_window_dims` are unique and sorted.
+
+  * (C10) For all i $\in$ [0, size(`update_window_dims`)), $0 \le$
+    `update_window_dims`[i] $\lt$ rank(`updates`[k]), for any k $\in$ [0, N).
+
+
+  * (C11) All dimensions in `inserted_window_dims` are unique and sorted.
+
+  * (C12)For all i $\in$ [0, size(`inserted_window_dims`)), $0 \le$
+    `inserted_window_dims`[i] $\lt$ rank(`inputs`[k]), for any k $\in$ [0, N).
+
+  * (C13) `inputs[k]` and `result[k]` have the same type for any k $\in$ [0, N).
+
+  * (C14) `update_computation` is of type `(I0, ..., IN-1, U0, ..., UN-1) -> (R0, ..., RN-1)`
+          where `I0, ..., IN-1` are types of `inputs`,  `U0, ..., UN-1` are
+          types of `updates`, and `R0, ..., RN-1` are types of `results`.
+
+
+  * On shape of `update`
+    * For all i $\in$ [0, N), rank(`updates[i]`) $=$ size(`update_window_dims`) +
+  rank(`scatter_indices`) - 1
+
+    * The dimension size of `i`$^{th}$ dimension, `di`, of `update` is computed
+      as:
+
+      * If `i` is present in `update_window_dims`, i.e., `i` $=$
+      `update_window_dims`[`k`] for some `k`, then `di` $=$
+      `adjusted_window_bounds`[`k`] where `adjusted_window_bounds` $=$
+      shape(`input`) - {dim(`input`, i) : i $\in$ `inserted_window_dims`} for
+      any `input` in `inputs`.
+
+      * If `i` is present in the `update_scatter_dims`, i.e., `i` $=$
+      `update_scatter_dims`[`k`] for some `k`, then
+        * `di` $=$ dim(`scatter_indices`, `k`) if `k` $\lt$ `index_vector_dim`
+        * `di` $=$ dim(`scatter_indices`, `k+1`), otherwise.
+
+### Examples
+
+```mlir
+// %operand: [[1.0, 4.0], [9.0, 25.0]]
+%result = "stablehlo.rsqrt"(%operand) : (tensor<2x2xf32>) -> tensor<2x2xf32>
```

Comment:
Why we have rsqrt example?

Reply:
Thanks for the catch! I uploaded the wrong one. Let me fix it.

Reply:
done.